laborator/content/reprezentare-numere/solution/2-len_xor/Makefile

```diff
@@ -0,0 +1,2 @@
+PROGNAME := len_xor
```

Comment:
On second thought, it would be better to just change this file to a link to `../../2-len_xor/Makefile`. This will make this makefile smaller and any changes to the skeleton makefile will automatically be replicated to the solution without losing functionality. Do the same for all the other makefiles in the `solution/` folder.

Reply:
Would this mean that after the required modifications, the Makefiles in the `solution/` directory will use the ones in the lab skeleton, which in turn use the generic in the common `utils/` directory, or am I getting this wrong?

Reply:
Yup, you're getting this right. The makefiles for the solutions will simply be symbolic links pointing to the ones from the skeletons, similar to how the lab folders [1] are links to subfolders in the `content/` directory [1].

[1] https://github.com/systems-cs-pub-ro/iocla/tree/master/laborator

Reply:
There seems to be a problem with this approach. While trying to implement this change, I get greeted with the following error message:

![image](https://user-images.githubusercontent.com/91455746/224545324-c48bdc1b-3d9c-457c-8910-3765915b8781.png)

This happens both when using soft-links or hard-links using the `ln` or `ln -s` command.

When calling `make`, the program will parse the contents of the file pointed to by the symlink, which specify that the `Makefile.generic` should be located at `../../utils/Makefile.generic`, relative to the `Makefile` in `lab/exercise/Makefile`, but when calling `make` from `lab/solution/exercise/Makefile` (symlink), `make` will think that this is the current path, and not `lab/exercise/Makefile`.

Is there a workaround to this problem ?

Reply:
The simplest solution would be to move the skeleton code in a `support` folder so that the paths from all Makefiles to the global ones are the same. @darius-m do you have a better one?

Reply:
If you want to add a `support` or `skel` directory for consistency with `sol`, I think that could be a good idea. But changing the directory structure just to be able to create symlinks to makefiles that contain two or three lines of variable definitions and includes is probably more work than it's worth, since updating just the makefile (without changing the task itself) is likely an uncommon occurance.

Reply:
You're right, @darius-m, that changing the structure for just a few lines that we are going to remove next semester when we move to OEH, is redundant. Before we merge this, I'll ask around first. Maybe there's a simpler solution.

Reply:
@RCX777 yeah it seems that using links is going to be too much of a hustle. Let's keep the current format.